CMakeLists.txt

```diff
@@ -352,7 +352,7 @@ if(SQLITE3MC_STATIC_RUNTIME_LINK)
 endif()
 message("Will build ${SQLITE3MC_TARGET} as ${SQLITE3MC_LINK}")
 
-set_target_properties(${SQLITE3MC_TARGET} PROPERTIES PUBLIC_HEADER "${SQLITE3MC_PUBLIC_HEADERS}")
+set_target_properties(${SQLITE3MC_TARGET} PROPERTIES PUBLIC_HEADER "${SQLITE3MC_PUBLIC_HEADERS};src/sqlite3mc_version.h")
```

Comment:
We will have to make `sqlite3mc_version.h` part of the list `SQLITE3MC_PUBLIC_HEADERS`, because it is included by `sqlite3mc.h`. Sorry, my fault - I simply missed this include file when I flagged the public header files in the list in issue #124.

Reply:
> This sqlite3mc_version.h is currently not added to the SQLITE3MC_TARGET as a source file.
> 
> * I can add it to SQLITE3MC_PUBLIC_HEADERS, but then it will be added as a source.

What's the problem with this approach? `sqlite3mc_version.h` **is** referenced from `sqlite3mc.h` and is therefore a **public** header file.

> * Or I can  append it to SQLITE3MC_PUBLIC_HEADERS, but only after the sources were added to SQLITE3MC_TARGET.

Why?

>       Which one should be?

IMHO it should be **added** to `SQLITE3MC_PUBLIC_HEADERS`.

Currently you have defined `SQLITE3MC_PUBLIC_HEADERS` as
```
set(SQLITE3MC_PUBLIC_HEADERS
  src/sqlite3.h
  src/sqlite3ext.h
  src/sqlite3mc.h
  src/sqlite3mc_vfs.h
  src/sqlite3userauth.h
)
```
because these are the header files a user of the library will need to be able to compile and build an application. `sqlite3mc_version.h` is included by `sqlite3mc.h` - that is, if it is missing compiling an application will fail. Therefore I think the right place to add it is `SQLITE3MC_PUBLIC_HEADERS`.

Reply:
Alright I'll add it to the public headers.
I was just wondering how did this compile so far. But I guess it's because the sqlite3mc_version.h was only used by sqlite3mc.h, and none of the other source files.

Reply:
> Alright I'll add it to the public headers.

Thanks.

> I was just wondering how did this compile so far. But I guess it's because the sqlite3mc_version.h was only used by sqlite3mc.h, and none of the other source files.

Well, `sqlite3mc.c` includes `sqlite3mc.h` which in turn includes `sqlite3mc_version.h`, and for example the version string (`SQLITE3MC_VERSION_STRING`) is used in function `sqlite3mc_version()` of `sqlite3mc.c` to return the version information. It compiles, because the header file resides in the same directory as the other header files and the source files.